src/runtime/analytics-service.js

```diff
@@ -74,6 +79,25 @@ export class AnalyticsService {
     this.lastAction_ = null;
   }
 
+  /**
+   * @param {boolean=} override
+   */
+  setTransactionId(override = false) {
+    if (override) {
+      this.transactionId_ = uuidFast();
+    }
+  }
+
+  /**
+   * return {string}
+   */
+  getTransactionId() {
+    if (!this.transactionId_) {
```

Comment:
But it's always defined and non-null. When is this possible then?

Reply:
Removed this check. This was again left-over from my original code where in the constructor, I would not set the value.